File: Weapon/GunshotWeapon/Pistol.php
<?php


namespace Weapon;


class Pistol extends GunshotWeapon
{
    private $armorPiercing;

    final public function getArmorPiercing(){
        return $this->armorPiercing;
    }
    final public function setArmorPiercing(float $armorPiercing){
        $this->armorPiercing = $armorPiercing;
    }
}

File: Weapon/GunshotWeapon/Pistol.py
from GunshotWeapon import GunshotWeapon


class Pistol(GunshotWeapon):
    __armorPiercing = None

    @property
    def armor_piercing(self):
        return self.__armorPiercing

    @armor_piercing.setter
    def armor_piercing(self, value: int):
        self.__armorPiercing = value
